### ISSUE TITLE
native integers don't match against Numeric

### ISSUE THREAD

**Issue by dwarring** [rakudo/rakudo#4485]
## The Problem

I'm surprised that int (and other native integers) don't match against Numeric, when Int is in it's MRO and does.
```
$ raku -e'say Int ~~ Numeric; say uint ~~ Int; say uint ~~ Numeric; say uint.^mro'
True
True
False
((uint) (Int) (Cool) (Any) (Mu))
```
Intended? should this be documented?

See also discussion on IRC https://colabti.org/irclogger/irclogger_log/raku?date=2021-08-04#l253

## Environment
* Operating system: ubuntu
* Compiler version (`perl6 -v` or `raku -v`): Welcome to 𝐑𝐚𝐤𝐮𝐝𝐨™ v2021.07-9-gffde2ba23.
Implementing the 𝐑𝐚𝐤𝐮™ programming language v6.d.
Built on MoarVM version 2021.07-8-g860cc6550.

**Comment by vrurg** [comment]
#4487 fixes the issue.

**Comment by vrurg** [comment]
Re-opening because turns out we don't have tests for natives type matching.

**Comment by vrurg** [comment]
Tests are implemented with Raku/roast#754
